src/data_provider/src/packages/packageLinuxParserApk.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015, Wazuh Inc.
+ * December 19, 2022.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "json.hpp"
+#include <fstream>
+
+void getApkInfo(const std::string& fileName, std::function<void(nlohmann::json&)> callback)
+{
+    std::ifstream apkDb(fileName);
+    const size_t valuePos {2};
+    std::string line;
+    std::vector<std::string> data;
+
+    if (apkDb.is_open())
+    {
+        while (getline(apkDb, line))
+        {
+            if (!line.empty())
+            {
+                switch (line.front())
```

Comment:
Why do all cases have the same implementation?

Reply:
The parsing is the same for all keys. I did a minor change to this.